pkgs/development/compilers/cudatoolkit/redist/find_outputs/find_outputs/utilities.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import subprocess
+import time
+from pathlib import Path
+from typing import Any, Iterable
+
+from pydantic import BaseModel, Field
+
+from find_outputs import manifest
+
+
+class NixStoreEntry(BaseModel):
+    hash: str
+    store_path: Path = Field(alias="storePath")
+
+
+def nix_store_prefetch_file(release: manifest.Release) -> NixStoreEntry:
+    """
+    Adds a release to the Nix store.
+
+    NOTE: By specifying the hash type and expected hash, we avoid redownloading.
+    """
+    url: str = f"{manifest.URL_PREFIX}/{release.relative_path}"
+    package_name: str = release.relative_path.split("/")[-1]
+    logging.debug(f"Adding {package_name} to the Nix store...")
+    start_time = time.time()
+    result = subprocess.run(
+        [
+            "nix",
+            "store",
+            "prefetch-file",
+            "--json",
+            "--hash-type",
+            "sha256",
+            "--expected-hash",
+            release.sha256,
+            url,
+        ],
+        capture_output=True,
+    )
+    result.check_returncode()
```

Comment:
easier to use `check=True`

Reply:
https://github.com/ConnorBaker/cuda-redist-find-features/commit/05765c20e502df7934fc1464a83dcfadcbc50b5d